source/quark/R/01_QuarkData.R

```diff
@@ -472,11 +481,71 @@ QuarkData$methods(
         data[ , targets] <<- tmp2
     },
 
-    cleanCollinVars = function(x)                                               {
+    
+   cleanCollinVars  = function(x)                                               {
         "Remove one variable from all collinear pairs"
-        collinVars <<- x
-        removeVars(x = unique(collinVars$var1), reason = "collinear")
-    },
+        collinVars     <<-  x
+        collinVarPairs <- collinVars[, 1:2]
+		
+        while(nrow(collinVarPairs) > 0){
+    
+        varCount     <- data.frame(table(unlist(collinVarPairs)))
+	maxVarCount  <- varCount[which(varCount$Freq == max(varCount$Freq)), ]
+	maxVarCommon <- intersect(names(respCounts), 
+		                          maxVarCount[, 1])
+    
+        ## Check for missing value counts if maximum counts are equal
+        if(nrow(maxVarCount) > 1){
+      
+        maxNaCountValue  <- max(respCounts[maxVarCommon])
+	maxNaVarNames    <- names(which(respCounts[maxVarCommon] == maxNaCountValue)) 
+        maxNaCount       <- data.frame(t(append(maxNaVarNames,
+		                                  maxNaCountValue)))
+										  
+        colnames(maxNaCount) <- c("var1", "count")
+      
+
+        ## Check if missing counts are same
+        if(nrow(maxNaCount) > 1){
+        
+        maxNaCount      <- maxNaCount[1, ]
+	maxNaCount      <- as.character(maxNaCount$var1)
+	collinVarPairs  <- subset(collinVarPairs, 
+		                          collinVarPairs[,1] != maxNaCount)
+	collinVarPairs  <- subset(collinVarPairs, 
+		                          collinVarPairs[,2] != maxNaCount)
+								  
+        sameNaCntVec    <- append(sameNaCntVec, maxNaCount)
+        
+        
+        }else if(nrow(maxNaCount) == 1){
+        
+        maxNaCount      <- as.character(maxNaCount$var1)
+	collinVarPairs  <- subset(collinVarPairs, 
+		                          collinVarPairs[,1] != maxNaCount)
+	collinVarPairs  <- subset(collinVarPairs, 
+		                          collinVarPairs[,2] != maxNaCount)
+								  
+        diffNaCntVec    <- append(diffNaCntVec, maxNaCount)
+         }
+      
+        ## Check if maximum counts are not equal
+        }else if(nrow(maxVarCount) == 1){
+      
+        firstVar        <- as.character(maxVarCount$Var1)
+	collinVarPairs  <- subset(collinVarPairs, 
+	                             collinVarPairs[,1] != firstVar)
+        collinVarPairs  <- subset(collinVarPairs, 
+	                             collinVarPairs[,2] != firstVar)
+								 
+        diffMaxCntVec   <- append(diffMaxCntVec, firstVar)
+                  }
+            }
+			
+        vector     <- c(sameNaCntVec,diffNaCntVec, diffMaxCntVec)
```

Comment:
'vector' is the name of a built-in R function (i.e., the one that creates vectors). We don't want to use extant function names as variable names. Please rename this variable.